### ISSUE TITLE
zfs pool created with FreeBSD not detected

### ISSUE THREAD

**Issue by superbonaci** [openzfsonwindows/ZFSin#230]
Operating system: Windows 10 build 18362.
OpenZFS installed: OpenZFSOnWindows-debug-20200206.exe

The pool was created as `tank` pool, then renamed to `zfs` pool. Then label was added to ad0 with glabel.

```
root@ubuntu:/# lsblk -o NAME,TYPE,FSTYPE,LABEL,SIZE,MODEL,MOUNTPOINT /dev/sdb
NAME TYPE FSTYPE     LABEL  SIZE MODEL          MOUNTPOINT
sdb  disk zfs_member zfs    3,9G ADATA
```
When attached to Windows:
```
C:\WINDOWS\system32>zpool list
no pools available
```
```
C:\WINDOWS\system32>zpool import
path '\\?\usbstor#disk&ven_xxxx&prod_xxxx_xxxxx&rev_x.xx#xxxxxxxxxxxxxxxx&0#{xxx}'
 and '\\?\PhysicalDrive4'
read partitions ok 1
    mbr 0: type 4 off 0x0 len 0xXXXXXXXXX
asking libefi to read label
[...]
no pools available to import
```

**Comment by lundman** [comment]
If it is on a partition, as opposed to whole-disk, then ZFSin will not yet "see" it. The partition choice for pools have not yet been implemented. It is assumed wholedisk, or file-based.

I'll see if I have some time to look at it. In fact, if you could make a small - mostly empty - pool the way you made it (that doesn't work) and send it to me, I'll have something to work on.

**Comment by superbonaci** [comment]
I've used the whole disk, "zpool create tank da1", not any partition inside. Will try to make the raw image and send it to you.

**Comment by superbonaci** [comment]
email sent

**Comment by lundman** [comment]
Downloaded, used vhdtool to convert .raw to .vhd, mounted as disk, and it imports fine:

```
  pool: zfs
 state: ONLINE
status: Some supported features are not enabled on the pool. The pool can
        still be used, but some features are unavailable.
action: Enable all features using 'zpool upgrade'. Once this is done,
        the pool may no longer be accessible by software that does not support
        the features. See zpool-features(5) for details.
  scan: none requested
config:

        NAME                   STATE     READ WRITE CKSUM
        zfs                    ONLINE       0     0     0
          /dev/physicaldrive3  ONLINE       0     0     0
```

Is this an actual USB stick? I perhaps should try that locally.

**Comment by superbonaci** [comment]
Yes, it's an USB stick. Only local disk are supported?

**Comment by superbonaci** [comment]
> Downloaded, used vhdtool to convert .raw to .vhd, mounted as disk, and it imports fine:

Hi again, did the same as you, downloaded VhdTool 1.0.0.1, then converted the dd file to vhd, then mounted it with Windows Disk Management, then `zpool.exe import zfs` worked fine.

But for some reason when the device is a usb stick doesn't detect it. Maybe any usb stick can do it, you can just create any test device for it.

I've mounted it read-write and works fine as fas a you use the command line to operate the filesystem. using Windows Explorer crashes Windows 10, will create another ticket for it.

**Comment by lundman** [comment]
It's meant to work, but clearly doesn't - in a VM it isn't always easy to test a physical USB stick, but I have a few ways to try.

**Comment by superbonaci** [comment]
I guess you will need a physical computer with Windows 10 installed, because I have more bugs to report. Virtualbox, QEMU and VMware Workstation have its bugs.

**Comment by superbonaci** [comment]
Ok tried myself with a virtual machine inside Virtualbox, linux host and Windows 10 guest, same thing as native W10 installation. When you connect the device to the virtual machine from `Virtualbox -> Devices -> USB -> usb-stick` doesn't detect the pool. You can test it on a virtual machine no need for native W10 install.

**Comment by lundman** [comment]
I'm remote from the vmware host as well, but I'll assemble some hardware here and give it a go. It is high time for me to do more hardware tests

**Comment by superbonaci** [comment]
Additional issue here: https://github.com/openzfsonwindows/ZFSin/issues/244

**Comment by lundman** [comment]
I've not forgotten this, but things are busy at the moment :)

**Comment by superbonaci** [comment]
Still happening with `OpenZFSOnWindows-debug-20200402.exe`
